martin/src/mbtiles/mod.rs

```diff
@@ -86,11 +87,19 @@ impl Source for MbtSource {
     }
 
     async fn get_tile(&self, xyz: &Xyz, _url_query: &Option<UrlQuery>) -> Result<Tile, Error> {
-        Ok(self
+        if let Some(tile) = self
```

Comment:
While we **could** change the `Source` trait to return an `Option<Tile>` instead of `Tile`, but in reality we already have that -- we can simply return an empty `Vec::default()` which has zero allocation cost (empty vectors don't cost anything).

This way it could be clearly indicated that a tile is simply missing, and no logging is done (we could still log it with debug level).

An error would then indicate all abnormal conditions like broken connection, IO error, etc, but not an empty tile.

Reply:
Maybe we could just drop the debug!("Couldn't find tile data in ..."), as it's too common that couldn't find tile data.

Reply:
sure, or at least switch it to `trace!`?

Reply:
P.S. Sorry, I didn't realize you were already using `Vec::new()` -- its the same as `Vec::default()`, and actually i prefer the `new()` because it is shorter